### PR TITLE
Update CHECKOUT_code

### PR DESCRIPTION
**Pull request by kaiyuan-cheng**
**Description**

SHiELD_build fails to compile the model out-of-the-box because outdated codes were checked out. Resolved the issue by revising the tags used for code checkout.

**How Has This Been Tested?**

Tested on Gaea and Stellar without issues.

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CHECKOUT_code
Comment (lharris4) [inline CHECKOUT_code:39]: It is good that we default to `main` for our base checkout, although it may be worthwhile to note in the README that this by default checks out the 'bleeding edge' code.

**Comment by laurenchilutti** [comment]
@lharris4 would you like the readme updated before we merge this in?

**Comment by lharris4** [comment]
> @lharris4 would you like the readme updated before we merge this in?

@laurenchilutti if we could; something like "By default, `CHECKOUT_code` checks out the latest main from each repository, which may include experimental features."

**Comment by laurenchilutti** [comment]
I was able to add a commit to update the README.  Please give this another review and I will merge when you approve.  Thank you!

**Review by bensonr — APPROVED** [review]

**Pull request merged**
